packages/preview-middleware-client/src/adp/quick-actions/fe-v4/utils.ts

```diff
@@ -1,53 +1,8 @@
-import type ManagedObject from 'sap/ui/base/ManagedObject';
-import type { Manifest } from 'sap/ui/rta/RuntimeAuthoring';
-import Component from 'sap/ui/core/Component';
-import type TemplateComponent from 'sap/fe/core/TemplateComponent';
-import type AppComponent from 'sap/fe/core/AppComponent';
-
-import { getControlById, isA } from '../../../utils/core';
+import { getControlById } from '../../../utils/core';
 import FlexCommand from 'sap/ui/rta/command/FlexCommand';
 import { QuickActionContext } from '../../../cpe/quick-actions/quick-action-definition';
 import CommandFactory from 'sap/ui/rta/command/CommandFactory';
-
-/**
- * Get the containing app component of a control.
- *
- * @param control - UI5 control instance.
- * @returns App component to which the control belongs.
- */
-export function getAppComponent(control: ManagedObject): AppComponent | undefined {
-    const ownerComponent = Component.getOwnerComponentFor(control);
-    if (isA<TemplateComponent>('sap.fe.core.TemplateComponent', ownerComponent)) {
-        return ownerComponent.getAppComponent();
-    }
-    return undefined;
-}
-/**
- * Get the containing page name of a control.
- *
- * @param control - UI5 control instance.
- * @returns Page name to which the control belongs.
- */
-export function getPageName(control: ManagedObject): string | undefined {
-    const component = Component.getOwnerComponentFor(control);
-    if (!isA<TemplateComponent>('sap.fe.core.TemplateComponent', component)) {
-        return undefined;
-    }
-    const view = component.getRootControl();
-    return view.getId().split('::').pop();
-}
-
-/**
- * Gets a reference id for a control.
- *
- * @param control - UI5 control instance.
- * @returns Reference id.
- */
-export function getReference(control: ManagedObject): string {
-    // probably same as flex setting id or base id TODO: CONFIRM
-    const manifest = getAppComponent(control)?.getManifest() as Manifest;
-    return manifest?.['sap.app']?.id ?? '';
-}
+import { getAppComponent, getPageName, getReference } from '../../../utils/fe-v4/utils';
```

Comment:
`utils` file in `utils` folder is a bit too much. Since there is only one file in the `fe-v4` why not just have a file with that name under `src/utils`. If it grows too much then we can think if we need to create another folder.

Reply:
moved